[AIRFLOW-328][AIRFLOW-371] Remove redundant default configuration & fix unit test configuration

jlowin opened the pull request:
AIRFLOW-328
https://issues.apache.org/jira/browse/AIRFLOW-328
Previously, Airflow had both a default template for airflow.cfg AND a
dictionary of default values. Frequently, these get out of sync (an
option in one has a different value than in the other, or isn’t present
in the other). This commit removes the default dict and uses the
airflow.cfg template to provide defaults. The ConfigParser first reads
the template, loading all the options it contains, and then reads the
user’s actual airflow.cfg to overwrite the default values with any new
ones.

AIRFLOW-371
https://issues.apache.org/jira/browse/AIRFLOW-371
Calling test_mode() didn't actually change Airflow's configuration! This actually wasn't an issue in unit tests because the unit test run script was hardcoded to point at the unittest.cfg file, but it needed to be fixed.

jlowin committed:
[AIRFLOW-328] Remove redundant default configuration

Previously, Airflow had both a default template
for airflow.cfg AND a dictionary of default
values. Frequently, these get out of sync (an
option in one has a different value than in the
other, or isn’t present in the other). This commit
removes the default dict and uses the airflow.cfg
template to provide defaults. The ConfigParser
first reads the template, loading all the options
it contains, and then reads the user’s actual
airflow.cfg to overwrite the default values with
any new ones.

r39132 commented on airflow/configuration.py:
```diff
@@ -311,7 +223,7 @@ def run_command(command):
 
 # Expose the configuration file in the web server
-expose_config = true
+expose_config = false
```
Should this be false or False?

jlowin replied:
It actually doesn't matter -- in order to work with it users have to call getbool() (otherwise it's a string), which loads any capitalization. But I will make it False for clarity.

r39132 commented on airflow/configuration.py:
```diff
@@ -551,18 +508,20 @@ def _get_env_var_option(self, section, key):
 
     def _get_cmd_option(self, section, key):
```
What is this used for?

jlowin replied:
There was a PR a while ago to allow certain configuration options to be specified as bash commands. That way passwords could be kept even more securely hidden, and retrieved via command.

jlowin committed:
[AIRFLOW-371] Make test_mode() functional

Previously, calling test_mode() didn’t actually
do anything.

This PR renames it to load_test_config() (to
avoid confusion, ht @r39132).

In addition, manually entering test_mode after
Airflow launches might be too late — some
options have already been loaded (DAGS_FOLDER,
etc.). This makes it so setting
tests/unit_test_mode OR the equivalent env var
(AIRFLOW__TESTS__UNIT_TEST_MODE) will load the
test config immediately, prior to loading the
rest of Airflow.
